### PR TITLE
Convert PostgreSQL system_identifier to uint64 and add tests

### PR DESCRIPTION
**Pull request by Mot1x**
# Pull request description

### Describe what this PR fixes
This pull request improves the handling and testing of PostgreSQL system identifiers, particularly focusing on correct type conversion from `int64` to `uint64` to handle edge cases such as negative values and large integers. The main changes include updating the assignment logic in the query runner and adding comprehensive tests to ensure correct conversion behavior.

**Bug fix and type handling:**

* Updated `getSystemIdentifier` in `query_runner.go` to first scan the system identifier as an `int64`, then safely convert it to `uint64` before assignment, ensuring correct handling of negative values and large numbers.

**Testing improvements:**

* Added `TestSystemIdentifierToUint64Y2K38` in `query_runner_test.go` to verify that various edge-case `int64` values (including negative, large, and minimum values) are correctly converted to `uint64`, preventing potential bugs with system identifier interpretation.

### Please provide steps to reproduce (if it's a bug)
from #2097:
1. Set system time to later than 03:14:07 on 19 January 2038.
2. Perform a backup: wal-g backup-push "$PGDATA" --full
3. See error:
```
INFO: 2038/01/18 19:15:38.030390 Backup will be pushed to storage: default
WARNING: 2038/01/18 19:15:38.041779 Couldn't get system identifier because of error: 'System Identifier: getting identifier of DB failed: can't scan into dest[0]: -9223371710917873611 is less than zero for uint64'
ERROR: 2038/01/18 19:15:38.042630 System Identifier: getting identifier of DB failed: can't scan into dest[0]: -9223371710917873611 is less than zero for uint64
```

### CODE REVIEW

File: internal/databases/postgres/query_runner_test.go [hunk 1]
```diff
@@ -81,3 +81,39 @@ func TestIsTablespaceMapExists(t *testing.T) {
...
+			input:    -1 << 63,
+			expected: 1 << 63,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := uint64(tc.input)
+			assert.Equal(t, tc.expected, actual)
+		})
```
[the remark this answers is not in the document]
Comment (Mot1x) [reply internal/databases/postgres/query_runner_test.go:117]: @copilot apply changes based on [this feedback](https://github.com/wal-g/wal-g/pull/2239#discussion_r3107077008)

**Comment by chipitsine** [comment]
@Mot1x  please mark as "resolved" items reviewed by you. 
also, do not treat copilot too seriously, it fails from time to time

**Comment by Mot1x** [comment]
@chipitsine review this. I marked this as resolved

**Review by serprex — APPROVED** [review]

**Comment by chipitsine** [comment]
@Mot1x can you have a look please https://github.com/wal-g/wal-g/actions/runs/25039230396/job/73338842409

**Comment by Mot1x** [comment]
@chipitsine next commit in master has 89/89 tests so there's no problem i guess
